### ISSUE TITLE
[Bug] Strimzi is wrongly copying well-known labels to managed objects

### ISSUE THREAD

**Issue by jpkrohling** [strimzi/strimzi-kafka-operator#2107]
**Describe the bug**
Operators are [recommended to label](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) its managed resources, like:

```yaml
  labels:
    app: jaeger
    app.kubernetes.io/component: kafka
    app.kubernetes.io/instance: auto-provision-kafka
    app.kubernetes.io/managed-by: jaeger-operator
    app.kubernetes.io/name: auto-provision-kafka
    app.kubernetes.io/part-of: jaeger
```

The example above is a `Kafka` instance created by the Jaeger Operator as part of a Jaeger instance named `auto-provision-kafka`.

The same set of labels are passed down verbatim to all components managed by the Kafka Operator. For instance, this is what we see as part of the Kafka statefuf set:

```yaml
  labels:
    app: jaeger
    app.kubernetes.io/component: kafka
    app.kubernetes.io/instance: auto-provision-kafka
    app.kubernetes.io/managed-by: jaeger-operator
    app.kubernetes.io/name: auto-provision-kafka
    app.kubernetes.io/part-of: jaeger
    strimzi.io/cluster: auto-provision-kafka
    strimzi.io/kind: Kafka
    strimzi.io/name: auto-provision-kafka-kafka
```

This causes problems with the Jaeger Operator, as it thinks that such resource is managed by itself. As it's not part of the manifest of resources that should be part of the instance, it ends up removing critical pieces, like secrets:

```
DEBU[0376] deleting secrets                              instance=auto-provision-kafka namespace=kafka secret=auto-provision-kafka-clients-ca-cert
DEBU[0376] deleting secrets                              instance=auto-provision-kafka namespace=kafka secret=auto-provision-kafka-cluster-ca
DEBU[0376] deleting secrets                              instance=auto-provision-kafka namespace=kafka secret=auto-provision-kafka-cluster-ca-cert
DEBU[0376] deleting secrets                              instance=auto-provision-kafka namespace=kafka secret=auto-provision-kafka-cluster-operator-certs
DEBU[0376] deleting secrets                              instance=auto-provision-kafka namespace=kafka secret=auto-provision-kafka-entity-operator-certs
DEBU[0376] deleting secrets                              instance=auto-provision-kafka namespace=kafka secret=auto-provision-kafka-kafka-brokers
DEBU[0376] deleting secrets                              instance=auto-provision-kafka namespace=kafka secret=auto-provision-kafka-zookeeper-nodes
DEBU[0376] deleting secrets                              instance=auto-provision-kafka namespace=kafka secret=auto-provision-kafka-clients-ca
```

A workaround on the Jaeger side is to use a different label than "managed-by" for the Kafka components (`Kafka` and `KafkaUser`), as that's how we internally use to compare "existing" vs. "expected" for all other components (like `Secret`).

**Expected behavior**
The Strimzi Operator should not copy over the values from well-known labels. At the very least, it should remove the "managed-by" label. 

**Environment (please complete the following information):**
 - Strimzi version: 0.14.0
 - Installation method: YAML
 - Kubernetes cluster: Kubernetes v1.15.2
 - Infrastructure: Minikube

**Comment by ppatierno** [comment]
Hi @jpkrohling , in the current master we have got merged this PR #2086 about having the operator setting the main labels (name, instance and managed-by) on all the resources handled by the operator itself. The "managed-by" is set to "strimzi-cluster-operator" which should already avoid the problem you are facing, right?

**Comment by jpkrohling** [comment]
Yes, it should fix the problem! Is there a way to test the current master?

**Comment by ppatierno** [comment]
You can install the CRDs and the cluster operator using the YAML files provided in the `install/cluster-operator` folder (on master branch). You need a user with admin rights on OpenShift.

**Comment by scholzj** [comment]
I think we need to check which labels will be overwritten by the CR labels and which wont. We didn't thought baout this use case when we did the #2086

**Comment by scholzj** [comment]
This should be now addressed and will be released as part of 0.15.0. Thanks for raising this.
